python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -1958,6 +1958,244 @@ keys:
+                      min: 100
+                      max: 9236
+                mac_address:
+                  type: dict
+                  keys:
+                    src:
+                      type: str
+                      pattern: "^([0-9a-f]{2}:){5}[0-9a-f]{2}$"
+                      description: MAC address for the source.
+                    dest:
+                      type: str
+                      pattern: "^([0-9a-f]{2}:){5}[0-9a-f]{2}$"
+                      description: MAC address for the destination.
+                encapsulation:
+                  type: dict
+                  keys:
+                    vxlan_strip:
+                      type: bool
+                      description: |-
+                        Strip VXLAN encapsulation header.
+                        `encapsulation.vxlan_strip` and `mpls_pop_all` are mutually exclusive.
+                        `mpls_pop_all` takes precedence.
+                    gre:
+                      type: dict
+                      keys:
+                        strip:
+                          type: bool
+                          description: Strip GRE encapsulation header without protocols.
+                        protocols:
+                          type: list
+                          primary_key: protocol
+                          description: Protocols for all destinations; destination-specific protocols should be set under the `destinations[].protocols` key.
+                          items:
+                            type: dict
+                            keys:
+                              protocol:
+                                type: str
+                                description: |-
+                                  Protocol type in GRE header.
+                                  Protocol range: 0x0000-0xFFFF.
+                              strip:
+                                type: bool
+                                description: |-
+                                  This is a required key to strip GRE encapsulation header with protocols.
+                              feature_header_length:
+                                description: |-
+                                  Feature header length in bytes.
+                                  This setting does not reflect in the EOS running-config for protocol 0x0000.
+                                type: int
+                                convert_types:
+                                  - str
+                                min: 1
+                                max: 16
+                              re_encapsulation_ethernet_header:
+                                type: bool
+                                description: |-
+                                  Extra ethernet header to prepend to the terminated packet.
+                                  This setting does not reflect in the EOS running-config for protocol 0x0000.
+                        destinations:
```

Comment:
`destinations` is not accurate key as we are giving both source and destination under this key. May be something like `routes` can be set.

Reply:
we try to keep the keys same as the CLI

Reply:
Use ```switchport tap encapsulation gre [tunnel destination <dst IP address> [source <src IP address> ]] strip``` to enable GRE termination on a tap port. The command allows specification of tunnel endpoints, to support termination on a specific tunnel.
switch(config-if-Et7/1)# switchport tap encapsulation gre destination <dst-ip> source <src-ip> strip

Reply:
Documentation says it is a tunnel with source and destination. So I think we rename this key to `tunnels`.

Reply:
eos cli also says it is a tunnel.

```s1-leaf1(config-if-Et2)#switchport tap encapsulation gre ?
  destination  Destination of tunnel packets```

Reply:
I am not sure, as we are mentioning `tunnel` destinations in the description below.
 @Vibhu-gslab @MaheshGSLAB @ClausHolbechArista @gmuloc any suggestions?

Reply:
looks ok to me as in description we covering the detail of key

Reply:
lets leave it with destinations